src/components/molecules/VenueJoinButton/VenueJoinButton.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { useCallback } from "react";
+import { VenueAccessMode } from "types/VenueAcccess";
+import { AnyVenue } from "types/venues";
+import { getCurrentTimeInUTCSeconds, getTimeBeforeParty } from "utils/time";
+import { joinVenue } from "utils/url";
+import { SecretPasswordForm } from "../SecretPasswordForm";
+
+interface VenueJoinButtonProps {
+  venueId?: string;
+  venue: AnyVenue;
+  onPasswordSubmit: () => void;
+  onPasswordSuccess: () => void;
+}
+
+export const VenueJoinButton: React.FC<VenueJoinButtonProps> = ({
+  venueId,
+  venue,
+  onPasswordSubmit,
+  onPasswordSuccess,
+}) => {
+  const handleJoinVenue = useCallback(() => {
+    joinVenue(venueId, venue?.entrance);
+  }, [venue, venueId]);
+
+  // @debt Handle emails and codes as well.
```

Comment:
Why is this `@debt`? Seems related to a core part of the PR/venue access feature?

Reply:
When the PR was created I was given UI only for password. I don't have any information about `codes` and `emails`.

Reply:
Hrmm, fair enough.

That's probably something that we may need to get from @sparkletown/design and/or @sparkletown/product 

Can you please sync up on that, and/or create a backlog ticket to track adding these parts.

Reply:
Yes, will do.

Reply:
Thankyou :)